### PR TITLE
Change logic of forceHttps

### PR DESCRIPTION
**Pull request by ZacZhang**
https://github.com/nanoPayinc/NANOPAY/issues/3890
https://github.com/foam-framework/foam2/pull/1589

Current Logic:
if (group.url exist) {
  config.url = group.url
}

else if (http request exist) {
  if (forceHttps && config.getUrl().startsWith("https")) {
    set https on request url
  }
  config.url = requestr url
}

else {
 use default config.url
}

### CODE REVIEW

**Comment by ronin00** [comment]
**NOTE: Do not merge.  Doesn't fully satisfy the points discussed here:**
https://github.com/foam-framework/foam2/pull/1624#issuecomment-431195443

**Comment by jlhughes** [comment]
This is what I want.

group.getAppConfig() {
  config = ctx get "appConfig"
  if requestUrl
    config.url = request url
  else if group.url 
    config.url = group.url
  
  if forceHttps
    set https on url

  return config

NOTE: test for leading http:// before stripping, we want to move to having URL just contain the domain name, without the protocol prefix.   Which will be the next PR, but it includes build/jenkins changes.

File: src/foam/nanos/auth/Group.js [hunk 1]
```diff
@@ -185,18 +185,25 @@ if ( session != null ) {
       //populate AppConfig url with request's RootUrl
       HttpServletRequest req = x.get(HttpServletRequest.class);
 
-      if ( req != null && ! SafetyUtil.isEmpty(req.getRequestURI()) ) {
+      if ( (req != null) && !SafetyUtil.isEmpty(req.getRequestURI()) ) {
```
Comment (jlhughes) [inline src/foam/nanos/auth/Group.js]: Why do we test for session above?  What happens when cron runs and there is no session?

Comment (ZacZhang) [reply src/foam/nanos/auth/Group.js]: Actually, we can remove the session check. 
I think session cannot be null, even for cron job, the session is 
screen shot 2018-10-22 at 11 00 20 am

Comment (jlhughes) [reply src/foam/nanos/auth/Group.js]: Great, one less thing.

File: src/foam/nanos/auth/Group.js [hunk 2]
```diff
@@ -173,38 +173,30 @@ DAO userDAO      = (DAO) x.get("localUserDAO");
...
-          config.setUrl(((Request) req).getRootURL().toString());
-        }
-      }
+      configUrl = group.getUrl();
     }
   }
 }
+
+if ( config.getForceHttps() && configUrl.startsWith("http://") ) {
+  configUrl = "https" + configUrl.substring(4);
```
Comment (jlhughes) [inline src/foam/nanos/auth/Group.js]: Need to handle the url being just the domain name like ‘portal.nanopay.net’ and with protocol: ‘http://portal.nanopay.net’.   If forceHttps is true, then both need to become ‘https://portal.nanopay.net’

Comment (ZacZhang) [reply src/foam/nanos/auth/Group.js]: updated code

**Review by jlhughes — APPROVED** [review]

**Pull request merged**